tests/snuba/sessions/test_sessions_v2.py

```diff
@@ -72,6 +72,75 @@ def test_virtual_groupby_query():
     assert query.query_groupby == []
 
 
+@freeze_time("2020-12-18T11:14:17.105Z")
+def test_massage_empty():
+    query = _make_query("statsPeriod=1d&interval=1d&field=sum(session)")
+
+    result_totals = []
+    result_timeseries = []
+
+    expected_result = {
+        "query": "",
+        "intervals": ["2020-12-18T00:00:00Z"],
+        "groups": [],
+    }
+
+    actual_result = result_sorted(massage_sessions_result(query, result_totals, result_timeseries))
+
+    assert actual_result == expected_result
+
+
+@freeze_time("2020-12-18T11:14:17.105Z")
+def test_massage_unbalanced_results():
+    query = _make_query("statsPeriod=1d&interval=1d&field=sum(session)&groupBy=release")
+
+    result_totals = [
+        {"release": "test-example-release", "sessions": 1},
+    ]
+    result_timeseries = []
+
+    expected_result = {
+        "query": "",
+        "intervals": ["2020-12-18T00:00:00Z"],
+        "groups": [
+            {
+                "by": {"release": "test-example-release"},
+                "series": {"sum(session)": [0]},
+                "totals": {"sum(session)": 1},
+            }
+        ],
+    }
+
+    actual_result = result_sorted(massage_sessions_result(query, result_totals, result_timeseries))
+
+    assert actual_result == expected_result
+
+    result_totals = []
+    result_timeseries = [
+        {
+            "release": "test-example-release",
+            "sessions": 1,
+            "bucketed_started": "2020-12-18T00:00:00+00:00",
+        },
+    ]
+
+    expected_result = {
+        "query": "",
+        "intervals": ["2020-12-18T00:00:00Z"],
+        "groups": [
+            {
+                "by": {"release": "test-example-release"},
+                "series": {"sum(session)": [1]},
+                "totals": {"sum(session)": 0},
```

Comment:
Will unbalanced results like this happen frequently? It looks odd to get value in the series, but have them not included in the totals.

Reply:
So the time series, and the totals are two different snuba queries which we do in series (we might be able to do those concurrently if you show me how), so there might be a few millis in between; enough time for consumers to produce more sessions. I think there might always be potential for this, more if you injest more sessions.